Add Loki Promtail (#688)
netgroup-polito/CrownLabs: infrastructure/monitoring/promtail/README.md

# Promtail

Promtail is an agent which ships the contents of local logs to a private Grafana Loki instance. It is usually deployed to every machine that has applications needed to be monitored.
It primarily:
 - Discovers targets
 - Attaches labels to log streams
 - Pushes them to the Loki instance

>  ## Prerequisites

Reviewer: ```suggestion
## Prerequisites
```

Helm must be installed to use the charts value. Please follow the [Helm installing guide](https://helm.sh/docs/intro/install/) before getting starting.

After Helm is ready, add the Grafana repo as follow:

> ```helm

Reviewer: Should be probably replaced with `bash` (same for the other occurrences)

helm repo add grafana https://grafana.github.io/helm-charts
helm repo update
```
To see the Grafana charts, you can run:

```helm
helm search repo grafana
```

> ## Installing

Reviewer: ```suggestion
## Installation
```

> To install the chart with the release name promtail and apply the configuration of values.yaml:

Reviewer: ```suggestion
To install the chart with the release name `promtail` and apply the configuration specified by the `values.yaml` file, it is possible to proceed as follows:
```

```helm

> helm upgrade --install promtail -f < YOUR_PATH >/values.yaml grafana/promtail

Reviewer: Would put the entire command, with the `--namespace monitoring --create-namespace` flags.
Would also put the path of the values file relative to the current directory, without the `< YOUR_PATH >` part.

```

> 

Reviewer: Would add a bit of introduction, saying that next you are presenting the most relevant modifications in the promtail configuration.

The [values.yaml](./values.yaml) is the file value for Promtail configuration.

```yaml
  snippets:
    pipelineStages:
      #- cri: {}
      - docker: {}
```

The **pipelineStages** is the way of extract data by parsing the log line. You can use the [Docker standard](https://grafana.com/docs/loki/latest/clients/promtail/stages/docker/), the [CRI standard](https://grafana.com/docs/loki/latest/clients/promtail/stages/cri/), a [regular expression](https://grafana.com/docs/loki/latest/clients/promtail/stages/regex/) or by parsing the log line as [JSON](https://grafana.com/docs/loki/latest/clients/promtail/stages/json/). You can only use one of these standard. 

> *NOTE: for JSON, you must configure **all** the expressions of your logs manually ([Using extraced data example](https://grafana.com/docs/loki/latest/clients/promtail/stages/json/#using-extracted-data)), because it does not recognize the key-value fields of the JSON, which means non-string elements like numbers, booleans or timestamp will not be assigne to those types.*

Reviewer: ```suggestion
*NOTE: for JSON, you must configure **all** the expressions of your logs manually ([Using extraced data example](https://grafana.com/docs/loki/latest/clients/promtail/stages/json/#using-extracted-data)), because it does not recognize the key-value fields of the JSON, which means non-string elements like numbers, booleans or timestamps will not be assigned to those types.*
```

### Bibliography

> https://grafana.com/docs/loki/latest/clients/promtail/stages/ <br>
> https://grafana.com/docs/loki/latest/clients/promtail/

Reviewer: ```suggestion
1. [Something](https://grafana.com/docs/loki/latest/clients/promtail/stages/)
2. [Promtail documentations](https://grafana.com/docs/loki/latest/clients/promtail/)
```

Nit. Seems just a bit better. Obviously, replace with the correct values.
